lib/transaction.js

```diff
@@ -81,12 +98,16 @@ Transaction.prototype.commit = function() {
     .sequelize
     .getQueryInterface()
     .commitTransaction(this, this.options)
-    .finally(function() {
+    .finally(function () {
       self.finished = 'commit';
       if (!self.options.transaction) {
         self.cleanup();
       }
-    });
+    })
+    .tap(
+      function (val) { self.promise.seqResolve(val); },
+      function (err) { self.promise.seqReject(err); }
```

Comment:
https://github.com/sequelize/sequelize/blob/master/lib/sequelize.js#L1156

`transaction.promise = new Promise()` and then return would solve the need for auto callback transactions.